pkg/policy/k8s/service_test.go

```diff
@@ -718,6 +730,40 @@ func TestPolicyWatcher_updateToServicesPoliciesTransformToEndpoint(t *testing.T)
 			svcByNameKey: {},
 		},
 	}, p.cnpByServiceID)
+
+	// test service has a selector label with a '.' in the key.
+	dottedCNP := &types.SlimCNP{
+		CiliumNetworkPolicy: &cilium_v2.CiliumNetworkPolicy{
+			TypeMeta:   metav1.TypeMeta{APIVersion: "cilium.io/v2", Kind: "CiliumNetworkPolicy"},
+			ObjectMeta: metav1.ObjectMeta{Name: "svc-dot-key", Namespace: "test"},
+			Spec:       &api.Rule{EndpointSelector: api.NewESFromLabels(), Egress: []api.EgressRule{{EgressCommonRule: api.EgressCommonRule{ToServices: []api.Service{{K8sService: &api.K8sServiceNamespace{ServiceName: "dot-svc", Namespace: "dot-ns"}}}}}}},
+		},
+	}
+	dottedKey := resource.NewKey(dottedCNP)
```

Comment:
I think it would be cleaner to extend existing test case instead of special casing this.

```go
svcByNameCNP := &types.SlimCNP{}

for _, tt := range []struct {
    svcSelector map[string]string
    
} {
    {
        svcSelector: map[string]string{
            "app": "foo",
        },
    },
    {
        svcSelector: map[string]string{
            "app.kubernetes.io": "foo",
        },
    }
} {
    servicesFixture.upsertService(fooSvcID, nil, tt.svcSelector, nil, nil)

    // Do all existing validation against the used tt.svcSelector.
    // .......

    expectedES := api.NewESFromK8sLabelSelector(labels.LabelSourceK8sKeyPrefix, &slim_metav1.LabelSelector{MatchLabels: tt.svcSelector})
	expectedES.Generated = true
}
```